test/acceptance/lookup-files.js

```diff
@@ -1,56 +1,79 @@
 var utils = require('../../lib/utils');
 
 describe('lookupFiles', function() {
-  var fs = require('fs'), path = require('path'), existsSync = fs.existsSync ||
-    path.existsSync;
+  var fs = require('fs'),
+    path = require('path'),
+    existsSync = fs.existsSync || path.existsSync,
+    tmpdir = require('os-tmpdir')(),
+    t = path.join.bind(path, tmpdir),
+    symlinkSupported = false;
+
+  fs.writeFileSync(t('mocha-utils.js'), 'yippy skippy ying yang yow');
+  try {
+    fs.symlinkSync(t('mocha-utils.js'), t('mocha-utils-link.js'));
+    symlinkSupported = true;
+  } catch (ignored) {
+  }
+
+  cleanup();
 
   beforeEach(function() {
-    fs.writeFileSync('/tmp/mocha-utils.js', 'yippy skippy ying yang yow');
-    fs.symlinkSync('/tmp/mocha-utils.js', '/tmp/mocha-utils-link.js');
+    fs.writeFileSync(t('mocha-utils.js'), 'yippy skippy ying yang yow');
+    if (symlinkSupported) {
+      fs.symlinkSync(t('mocha-utils.js'), t('mocha-utils-link.js'));
+    }
   });
 
-  it('should not choke on symlinks', function() {
-    expect(utils.lookupFiles('/tmp', ['js'], false))
+  (symlinkSupported ? it : it.skip)('should not choke on symlinks', function() {
+    expect(utils.lookupFiles(tmpdir, ['js'], false))
       .to
-      .contain('/tmp/mocha-utils-link.js')
+      .contain(t('mocha-utils-link.js'))
       .and
-      .contain('/tmp/mocha-utils.js')
+      .contain(t('mocha-utils.js'))
       .and
       .have
       .length(2);
-    expect(existsSync('/tmp/mocha-utils-link.js'))
+    expect(existsSync(t('mocha-utils-link.js')))
       .to
       .be(true);
-    fs.renameSync('/tmp/mocha-utils.js', '/tmp/bob');
-    expect(existsSync('/tmp/mocha-utils-link.js'))
+    fs.renameSync(t('mocha-utils.js'), t('bob'));
+    expect(existsSync(t('mocha-utils-link.js')))
       .to
       .be(false);
-    expect(utils.lookupFiles('/tmp', ['js'], false))
+    expect(utils.lookupFiles(tmpdir, ['js'], false))
       .to
       .eql([]);
   });
 
   it('should accept a glob "path" value', function() {
-    expect(utils.lookupFiles('/tmp/mocha-utils*', ['js'], false))
+    var res = utils.lookupFiles(t('mocha-utils*'), ['js'], false)
+      .map(path.normalize.bind(path));
+    var ex = expect(res)
       .to
-      .contain('/tmp/mocha-utils-link.js')
-      .and
-      .contain('/tmp/mocha-utils.js')
-      .and
+      .contain(t('mocha-utils.js'));
+
+    if (symlinkSupported) {
+      ex = ex.and
+        .contain(t('mocha-utils-link.js'));
+    }
+
+    ex.and
       .have
-      .length(2);
+      .length(1 + (+symlinkSupported));
   });
 
```

Comment:
For the expected length, I'd rather see something like:
```js
// at start of test
var expectedLength = 0;
//...after the .contain(t('mocha-utils.js'));
expectedLength += 1;
//...after the .contain(t('mocha-utils-link.js'));
expectedLength += 1;
//...instead of .length(1 + (+symlinkSupported));
.length(expectedLength);
```
This is, in my opinion, clearer than using the unary `+` to cast the `symlinkSupported` bool to the integer 1, and more flexible in the event this test were ever to be changed such that there would be more than one additional file added if `symlinkSupported`. Although a simpler alternative that requires manual counting (instead of adding a `+= 1` line to everywhere the count should increase) would be to just change `.length(1 + (+symlinkSupported));` to `.length(1 + (symlinkSupported ? 1 : 0));` or `.length(symlinkSupported ? 2 : 1);`.